src/ServiceControl.Config/Commands/UpgradeInstanceCommand.cs

```diff
@@ -28,13 +28,19 @@ public UpgradeInstanceCommand(Func<InstanceDetailsViewModel, bool> canExecuteMet
         public UpgradeInstanceCommand(IWindowManagerEx windowManager, IEventAggregator eventAggregator, Installer installer)
         {
             this.windowManager = windowManager;
-
             this.eventAggregator = eventAggregator;
             this.installer = installer;
         }
 
         public override async Task ExecuteAsync(InstanceDetailsViewModel model)
         {
+            var licenseCheckResult = installer.CheckLicenseIsValid();
+            if (!licenseCheckResult.Valid)
+            {
+                windowManager.ShowMessage(licenseCheckResult.Message, "Upgrade could not continue due to an issue with the current license. Contact sales@particular.net", hideCancel: true);
```

Comment:
@gbiellem, it seems the error message `licenseCheckResult.Message` is actually displayed as the title of the popup dialog, shouldn't the `licenseCheckResult.Message` be part of the body ?